src/probnum/filtsmooth/timeseriesposterior.py

```diff
@@ -36,9 +36,29 @@ class TimeSeriesPosterior(abc.ABC):
         Posterior random variables.
     """
 
-    def __init__(self, locations: np.ndarray, states: np.ndarray) -> None:
-        self.locations = np.asarray(locations)
-        self.states = _randomvariablelist._RandomVariableList(states)
+    def __init__(
+        self,
+        locations: Optional[Iterable[FloatArgType]] = None,
+        states: Optional[Iterable[randvars.RandomVariable]] = None,
+    ) -> None:
+        self._locations = list(locations) if locations is not None else []
+        self._states = list(states) if states is not None else []
+
+    def append(
+        self,
+        location: FloatArgType,
+        state: randvars.RandomVariable,
+    ) -> None:
+        self._locations.append(location)
+        self._states.append(state)
+
+    @property
+    def locations(self):
+        return np.asarray(self._locations)
+
+    @property
```

Comment:
do you think it makes sense to cache those? And does it also make sense to use (i) check in the append() whether the location is right of the last location (ii) add a finalise() method that blocks the append() for the future? I am not sure whether these things are good or bad ideas, but would like to hear your opinion :)

Reply:
> do you think it makes sense to cache those? 

I don't know what happens then when the object is altered (say, by appending another location+state)

(i): 
Mh technically, I would agree. But isn't that checked elsewhere? Resp., does the algorithm, which calls this method, even allow this?

(ii)
Absolutely, good idea. I'll add that. Do we want another class (e.g. `FrozenFilteringPosterior` ) for that? Or just a flag?

Reply:
> I don't know what happens then when the object is altered (say, by appending another location+state)
I guess it remains cached. But I am not 100% sure (though, isn't the idea of caching to not check these things anymore?)

(I) the method does not do this, but what happens if a user randomly decides to append a new measurement. This would be possible (though stupid? I don't know). I don't think it is really checked, but perhaps there are some weird errors. I don't know. I think it would be reasonable to check.

(ii) I think a flag would work, unless the FrozenFilteringPosterior makes the code more elegant. Do what you consider most simple :)

Reply:
I used a flag now, since I didn't find an elegant way to transfer the changes of a `FrozenTimeSeriesPosterior` to subclasses of `TimeSeriesPosterior` , like `FilteringPosterior` for instance. Before creating a hundred subclasses, I chose the easier way. Although it would have been nice to convert the property into a cached_property after freezing.

Reply:
Alright! 

> Although it would have been nice to convert the property into a cached_property after freezing

Technically, one can always manually cache it :) But I think it is okay without

Reply:
Only if you know beforehand how many states/locations you will add. But that's a good point, we could make that optionally possible. Shall we?

Sth like
```python
def __init__(..., total=None):
  if total is not None:
    self.locations = np.empty((total, ))
  ...
  # etc.
```

Reply:
Don't know how much we gain by this, though. I suspect not much, tbh